src/script/builder/tir/stmt.h

```diff
@@ -0,0 +1,307 @@
+    TIRFrameNode::VisitAttrs(v);
+    v->Visit("condition", &condition);
+  }
+
+  static constexpr const char* _type_key = "script.builder.tir.WhileFrame";
+  TVM_DECLARE_FINAL_OBJECT_INFO(WhileFrameNode, TIRFrameNode);
+
+ public:
+  void ExitWithScope() final;
+};
+
+class WhileFrame : public TIRFrame {
+ public:
+  TVM_DEFINE_MUTABLE_NOTNULLABLE_OBJECT_REF_METHODS(WhileFrame, TIRFrame, WhileFrameNode);
+};
+
+class IfFrameNode : public TIRFrameNode {
+ public:
+  PrimExpr condition;
+  Array<tvm::tir::Stmt> then_stmt;
+  Array<tvm::tir::Stmt> else_stmt;
+
+  void VisitAttrs(tvm::AttrVisitor* v) {
+    TIRFrameNode::VisitAttrs(v);
+    v->Visit("condition", &condition);
+    v->Visit("then_stmt", &then_stmt);
+    v->Visit("else_stmt", &else_stmt);
+  }
+
+  static constexpr const char* _type_key = "script.builder.tir.IfFrame";
+  TVM_DECLARE_FINAL_OBJECT_INFO(IfFrameNode, TIRFrameNode);
+
+ public:
+  void ExitWithScope() final;
+};
+
+class IfFrame : public TIRFrame {
+ public:
+  TVM_DEFINE_MUTABLE_NOTNULLABLE_OBJECT_REF_METHODS(IfFrame, TIRFrame, IfFrameNode);
+};
+
+class ThenFrameNode : public TIRFrameNode {
+ public:
+  static constexpr const char* _type_key = "script.builder.tir.ThenFrame";
+  TVM_DECLARE_FINAL_OBJECT_INFO(ThenFrameNode, TIRFrameNode);
+
+ public:
+  void ExitWithScope() final;
+};
+
+class ThenFrame : public TIRFrame {
+ public:
+  TVM_DEFINE_MUTABLE_NOTNULLABLE_OBJECT_REF_METHODS(ThenFrame, TIRFrame, ThenFrameNode);
+};
+
+class ElseFrameNode : public TIRFrameNode {
+ public:
+  static constexpr const char* _type_key = "script.builder.tir.ElseFrame";
+  TVM_DECLARE_FINAL_OBJECT_INFO(ElseFrameNode, TIRFrameNode);
+
+ public:
+  void ExitWithScope() final;
+};
+
+class ElseFrame : public TIRFrame {
+ public:
+  TVM_DEFINE_MUTABLE_NOTNULLABLE_OBJECT_REF_METHODS(ElseFrame, TIRFrame, ElseFrameNode);
+};
+
+tvm::tir::IterVar EnvThread(String thread_tag);
+void BufferStore_(tvm::tir::Buffer buffer, PrimExpr value, Array<PrimExpr> indices);
+void Prefetch_(tvm::tir::Buffer buffer, Array<Range> bounds);
+void Seq(Array<tvm::tir::Stmt> seq);
+void Evaluate_(PrimExpr value);
+
+AssertFrame Assert(PrimExpr condition, String message);
+LetFrame Let(tvm::tir::Var var, PrimExpr value);
+AllocateFrame Allocate_(Array<PrimExpr> extents, DataType dtype, String storage_scope_str = "",
+                        PrimExpr condition = true,
+                        Optional<Map<String, ObjectRef>> annotations = NullOpt);
+AllocateConstFrame AllocateConst_(tvm::runtime::NDArray data, DataType dtype,
+                                  Array<PrimExpr> extents);
+LaunchThreadFrame LaunchThread(tvm::tir::IterVar env_var, PrimExpr extent);
+RealizeFrame Realize(tvm::tir::BufferRegion buffer_slice, String storage_scope_str,
+                     PrimExpr condition);
+AttrFrame Attr(ObjectRef node, String attr_key, PrimExpr value);
+WhileFrame While_(PrimExpr condition);
+IfFrame If_(PrimExpr condition);
+ThenFrame Then_();
+ElseFrame Else_();
```

Comment:
Looks like we don't need the underscore in C++?

```suggestion
AssertFrame Assert(PrimExpr condition, String message);
LetFrame Let(tvm::tir::Var var, PrimExpr value);
AllocateFrame Allocate(Array<PrimExpr> extents, DataType dtype, String storage_scope_str = "",
                        PrimExpr condition = true,
                        Optional<Map<String, ObjectRef>> annotations = NullOpt);
AllocateConstFrame AllocateConst_(tvm::runtime::NDArray data, DataType dtype,
                                  Array<PrimExpr> extents);
LaunchThreadFrame LaunchThread(tvm::tir::IterVar env_var, PrimExpr extent);
RealizeFrame Realize(tvm::tir::BufferRegion buffer_slice, String storage_scope_str,
                     PrimExpr condition);
AttrFrame Attr(ObjectRef node, String attr_key, PrimExpr value);
WhileFrame While(PrimExpr condition);
IfFrame If(PrimExpr condition);
ThenFrame Then();
ElseFrame Else();
```

Reply:
Here I wonder whether it is conflict with those `stmt` in `tvm::tir`, since I noticed that you had named `Buffer_` for `BufferFrame`.